platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml

```diff
@@ -39,12 +33,184 @@
+    - network['type'] == 'indy' and item.services.stewards is defined
+
+- name: Get Elastic IP
+  environment:
+    AWS_ACCESS_KEY_ID: "{{ aws.access_key }}"
+    AWS_SECRET_ACCESS_KEY: "{{ aws.secret_key }}"
+  shell: |
+    # format ip addresses list to string with space separator
+    ips=$(echo '{{ item.publicIps }}' | tr -d '["]' | sed 's/,/\ /g')
+    data=$(aws ec2 describe-addresses --public-ips ${ips} --region {{ aws.region }} --output json | jq '.Addresses[].AllocationId')
+    # format eip addresses list to string with comma separator (comma has to be escaped)
+    echo ${data} | tr -d '"' | sed 's/\ /\,/g'
+  register: allocation_ips
+  when:
+     - ( network.type == 'indy' and item.services.stewards is defined )
+     - item.cloud_provider == 'aws' or item.cloud_provider == 'aws-baremetal' #As this shell code is specific to AWS, cloud provider check has been added
+  tags:
+    - notest
+
+- name: Set Helm value for ambassador_eip
+  set_fact:
+    allocation_ips_stdout: "{{ allocation_ips.stdout | default('No value', true) }}"
+  when:
+    - allocation_ips is defined
+
+- name: Extract port numbers from the 'ports' string
+  set_fact:
+    extracted_ports: "{{ ports | regex_replace('[^0-9,]', ' ') | split(',') | map('int') | list | default('No value', true) }}"
+  vars:
+    ports: "{{ network.env.ambassadorPorts.ports | default('') }}"
+  register: ambassadorports_list
+  when:
+    - network.type == 'indy'
+
+- name: Create custom values for aes helm chart
+  vars:
+    ports: "{{ network.env.ambassadorPorts.ports | default([]) }}"
+    lbSourceRangeDefault:
+      - 0.0.0.0/0
+    loadBalancerSourceRanges: "{{ network.env.loadBalancerSourceRanges | default(lbSourceRangeDefault) }}"
+    elastic_ip: "{{ allocation_ips_stdout }}"
+    extracted_ports: "{{ ambassadorports_list }}"
+  template:
+    src={{ playbook_dir }}/../../../platforms/shared/configuration/roles/setup/edge-stack/templates/aes-custom-values.tpl
+    dest={{ playbook_dir }}/../../../platforms/shared/configuration/build/aes-custom-values.yaml
+  when:
+    - network.type == 'indy'
+    
+- name: Format ambassador range
+  args:
+    executable: /bin/bash
+  shell: |
+    from='{{ network.env.ambassadorPorts.portRange.from | default('') }}'
+    to='{{ network.env.ambassadorPorts.portRange.to | default('') }}'
+    if [ -z "$from" ] || [ -z "$to" ] 
+    then
+      echo ""
+    else
+      echo "--set ambassador.otherPorts.portRange.from=${from} --set ambassador.otherPorts.portRange.to=${to}"
+    fi
+  register: ambassadorRange
+
+
+- name: Format ambassador ports for Indy
+  args:
+    executable: /bin/bash
+  shell: |
+    ports=$(echo '{{ network.env.ambassadorPorts.ports | default('') }}' | sed -e 's/\[/\{/' -e 's/\]/\}/')
+    terminalPorts='{{ terminal.stdout | default('') }}'
+    if [ -z "$ports" ] 
+    then
+      echo ""
+    else
+      echo "--set ambassador.otherPorts.ports={'${ports},${terminalPorts}'}"
+    fi
+  register: ambassadorPortsIndy
```

Comment:
what is the use of ambassadorPortsIndy ?

Reply:
removed ambassadorPortsIndy